Add audio text to text task

MrShahzebKhoso opened the pull request:
This PR introduces support for the Audio-Text-to-Text task in huggingface.js.

- Added details to the sections of the audio-text-to-text task section in the packages/tasks/src/tasks/audio-text-to-text/ directory that contains about.md and data.ts. 
- Ensured consistency with existing task structure and documentation.

merveenoyan commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,44 @@
+import numpy as np
+import librosa
+
+pipe = transformers.pipeline(model='fixie-ai/ultravox-v0_5-llama-3_2-1b', trust_remote_code=True)
```
would be nice to pick an officially implemented model like Voxtral

merveenoyan reviewed (commented):
thanks a lot! 
only about.md part needs a bit of changes, we try to keep these pages high quality sources 
also cc @Vaibhavs10 if you have time would be nice if you could review!

MrShahzebKhoso commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,44 @@
+
+pipe = transformers.pipeline(model='fixie-ai/ultravox-v0_5-llama-3_2-1b', trust_remote_code=True)
+
+path = "<path-to-input-audio>"  # TODO: pass the audio here
```
Added more examples that load from the URL. Checked on Google Colab and these run without needing anything else.

MrShahzebKhoso commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
````diff
@@ -0,0 +1,44 @@
+pipe({'audio': audio, 'turns': turns, 'sampling_rate': sr}, max_new_tokens=30)
+
+
+```
````
Added research papers, datasets, and code and demo links to the about.md file.

MrShahzebKhoso commented:
@merveenoyan Thank you very much for the guidance and valuable feedback. I have made all the changes. 

1. Updated inference section. Added 3 different codes that can be run directly. The files are from URLs. 
2. Added the Voxtral Examples in the inference section for more authenticity. 
3. Added more details and more use cases for better understanding.
4. Added resources section in the about.md file. This contains links to the papers, blogs, datasets, and codes. These will be helpful to anyone trying to explore more.

Kindly let me know if you have any further suggestions. I really appreciate your time and input!

Vaibhavs10 reviewed (commented):
gentle reminder @merveenoyan @MrShahzebKhoso to look at existing open PRs before opening a new one, in this case there was already an old task PR from @Deep-unlearning: https://github.com/huggingface/huggingface.js/pull/1212

That said, let's proceed with this one since it's from the community vs in-house.

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+## Use Cases
+
+> This task takes **audio** and an optional **text prompt** and returns **text** (answers, summaries, structured notes, etc.).
```
```suggestion
> This task takes `audio` and a `text prompt` and returns `text` (answers, summaries, structured notes, etc.).
```

Vaibhavs10 replied:
it's not optional, text prompt is always required in Audio LMs.

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+Go beyond transcription to extract intent, sentiment, uncertainty, or emotion from spoken language.
+**Example:** “I’m not sure I can finish this on time.” → Prompt: “Describe speaker intent and confidence.”
+
+### Lecture/podcast summarization with follow-ups
+Produce a brief summary, then answer follow-up questions that reference the audio.
+**Example:** Audio: 30-min podcast → Prompt: “Give a 5-bullet summary, then explain the guest’s main argument.”
+
```
```suggestion

```

Let's remove this since it's quite similar to second point with `Meeting notes & action items` - we can update that with details of this point in a concise manner.

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+## Inference
+You can use 'transformers' library, and your audio file to any of the `audio-text-to-text` model, with instructions and get text responses. Following code examples show how to do so.
+
+### Summarization / Q&A on a Single Audio
```
could you maybe take the inference examples from here: https://github.com/huggingface/huggingface.js/pull/1212 

just those two are sufficient.

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
````diff
@@ -0,0 +1,184 @@
+print("=" * 80)
+```
+
+## Useful Resources
````
similarly for the resources you can also enrich this list with the ones in this PR: https://github.com/huggingface/huggingface.js/pull/1212 (the more resources the better)

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+If you want to learn more about this concept, here are some useful links:
+
+### Papers
+- [SpeechGPT (Paper)](https://arxiv.org/pdf/2507.13264)
```
```suggestion
- [SpeechGPT (Paper)](https://huggingface.co/papers/2507.13264)
```

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+
+### Papers
+- [SpeechGPT (Paper)](https://arxiv.org/pdf/2507.13264)
+- [Voxtral (Paper)](https://arxiv.org/pdf/2507.13264)
```
```suggestion
- [Voxtral (Paper)](https://huggingface.co/papers/2507.13264)
```

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+### Papers
+- [SpeechGPT (Paper)](https://arxiv.org/pdf/2507.13264)
+- [Voxtral (Paper)](https://arxiv.org/pdf/2507.13264)
+- [Qwen2-audio-instruct (Paper)](https://arxiv.org/pdf/2407.10759)
```
```suggestion
- [Qwen2-audio-instruct (Paper)](https://huggingface.co/papers/2407.10759)
```

Vaibhavs10 commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,184 @@
+- [SpeechGPT (Paper)](https://arxiv.org/pdf/2507.13264)
+- [Voxtral (Paper)](https://arxiv.org/pdf/2507.13264)
+- [Qwen2-audio-instruct (Paper)](https://arxiv.org/pdf/2407.10759)
+- [AudioPaLM (Paper)](https://arxiv.org/pdf/2306.12925)
```
```suggestion
- [AudioPaLM (Paper)](https://huggingface.co/papers/2306.12925)
```

Vaibhavs10 reviewed (commented):
took a first pass, cc: @Deep-unlearning - could you take a look by Monday as well please.

pcuenca reviewed (approved):
LGTM with @Vaibhavs10's comments and the linter changes.

MrShahzebKhoso commented:
Thank you, @Vaibhavs10  and @pcuenca, for reviewing and  considering this PR. I’ve made the requested updates and pushed the changes.

1. Added the two examples from #1212.
2. Updated the resources section with additions from #1212.
3. Revised about.md to align with task documentation standards.
4. Ensured consistency in formatting and descriptions.

Looking forward to your feedback and approval.

Deep-unlearning commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,129 @@
+### Audio Question Answering 
+These models can understand audio directly and answer questions about it. For example, summarizing a podcast clip or explaining parts of a recorded conversation.
+
+You can experiment wit [Qwen2-Audio-Instruct-Demo](https://huggingface.co/Qwen/Qwen2-Audio-Instruct-Demo) for conversations with both text and audio inputs, letting you ask follow-up questions about different sounds or speech clips.
```
```suggestion
You can experiment with [Qwen2-Audio-Instruct-Demo](https://huggingface.co/Qwen/Qwen2-Audio-Instruct-Demo) for conversations with both text and audio inputs, letting you ask follow-up questions about different sounds or speech clips.
```

Deep-unlearning replied:
```suggestion
You can experiment wit [Qwen2-Audio-Instruct-Demo](https://huggingface.co/Qwen/Qwen2-Audio-7B-Instruct) for conversations with both text and audio inputs, letting you ask follow-up questions about different sounds or speech clips.
```

Deep-unlearning commented on packages/tasks/src/tasks/audio-text-to-text/about.md:
```diff
@@ -0,0 +1,129 @@
+**Example:** Song demo → Prompt: “Identify key and tempo, then suggest jazz reharmonization ideas for the chorus.”
+
+## Inference
+You can use 'transformers' library, and your audio file to any of the `audio-text-to-text` model, with instructions and get text responses. Following code examples show how to do so.
```
```suggestion
You can use the 'transformers' library, and your audio file to any of the `audio-text-to-text` model, with instructions and get text responses. Following code examples show how to do so.
```

MrShahzebKhoso commented:
Hi @Deep-unlearning, thank you for reviewing and the fixes. Really appreciate the support.
I have committed these changes.

Vaibhavs10 reviewed (approved):
Sorry for the delay here @MrShahzebKhoso, one last thing can you please run the linter on this:

```
cd huggingface.js

# to make sure to install latest deps
git checkout main
git pull 
pnpm i

cd packages/tasks
pnpm format
```
